openlcsd/flow/tasks.py

```diff
@@ -366,6 +368,44 @@ def download_shared_remote_source(context, engine):
         )
 
 
+def download_maven_source_from_corgi(context, engine, tmp_dir):
+    # deal with MAVEN component
+    component = context.get('component')
+    download_url = component.get('download_url')
+    if not download_url:
+        raise RuntimeError(f"component {component['purl']} download_url empty")
+    name_list = list_http_files(download_url)
+    pom_filename, jar_filename, sources_jar_filename = None, None, None
+    for name in name_list:
+        if name.endswith(".pom"):
+            pom_filename = name
+        if name.endswith(".jar") and "doc" not in name:
+            jar_filename = name
+        if name.endswith("sources.jar"):
+            sources_jar_filename = name
```

Comment:
Does this mean there would be no more than 1 pom/jar/sources.jar found in the download_url of maven component?

Reply:
I am not sure if there some component have multiple pom/jar/sources.jar. Do u have some example.